### ISSUE TITLE
Intent to fork - maintainers attention required

### ISSUE THREAD

**Issue by romgrk** [exceljs/exceljs#2764]
I work at [MUI](https://mui.com/) and we're using exceljs in our [data grid](https://github.com/mui/mui-x). We have an issue with an [outdated exceljs dependency](https://github.com/mui/mui-x/issues/13187) causing blocking security warnings for our users, as well as other unanswered issues like [this one](https://github.com/exceljs/exceljs/issues/713), and some new browser APIs could replace bundle-heavy modules in the frontend.

We need to make those changes but we haven't seen any maintainer activity in the last months. If there is still an active maintainer, would it be possible to confirm the status of this package? We're thinking about forking it, but if you're interested we could make the changes here directly, either through PRs or if you give us accesses we can manage the updates ourselves. We are already maintaining a large open-source project so we have experience. We can't promise much bandwidth but as our public API is tied to exceljs at the moment, we will be interested in keeping this package minimally maintained.

**Comment by romgrk** [comment]
cc @Siemienik, you seem to be the only active maintainer

**Comment by hyperliskdev** [comment]
I am using both MUI and ExcelJS for a contract project and I am not a maintainer but if there is interest in booting this project up again, I have some changes to make with the rendering of xml. It seems that some of the things like r:ids are missing from attributes and I have since been working on an issue with autoFilter breaking sheets upon write. I would be interested in helping make this project closer to the OOXML standard and as a cybersecurity student, I think it would be interesting to see how the security warnings are fixed. 

#2585

**Comment by hyperliskdev** [comment]
https://github.com/hyperliskdev/exceljs

I have started creating issues and fixing small issue as of now and I want to get the tests/build status back to working. I would be comfortable with y'all forking my repo and then I would definitly contribute to that one for MUI.

**Comment by Blackhol3** [comment]
There is indeed dozens of reported issues and PR waiting for attention. It would be a shame to let such a good library go to waste… Any update, @Siemienik?

**Comment by timothystegall** [comment]
Agreed, would appreciate someone (whether MUI or someone else) not letting this great library fall potentially unmaintained.

**Comment by kanapka94** [comment]
@Siemienik is this project still supported?

**Comment by mikez** [comment]
@romgrk I recently took a look at this project amidst contributing [pivot tables](https://github.com/exceljs/exceljs/pull/2551) for a client.

Here are the two main issues I see at this time:

1. **Underfunding**. Many people and companies are benefiting from this library, but few are paying for the 100s of hours that it takes review PRs, bugs, and leadership.
2. **Code expertise**. It's easy to make one-off fixes or a fork. It's hard to keep the overall code consistent and clean so it works for a wide group of users. Right now, there are only two people with in-depth knowledge of the code: @guyonroche (the main architect from 2014 until recently) and @Siemienik (the current owner, maintainer, and contributor since 2019). In addition to JavaScript expertise, you also need deep knowledge of the [Office Open XML standard](https://en.wikipedia.org/wiki/Office_Open_XML).

I believe the best way you can help right now is to **secure funding or sponsorship**; and help gather a team of experienced devs who can triage bugs and move this project forwards.

**Comment by hyperliskdev** [comment]
#2585 On this issue that I was working on one member mentioned some support in both funding another experienced dev.

**Comment by hyperliskdev** [comment]
https://github.com/hyperliskdev/ooxml-ts based on this project, I have started a new typescript library that will handle the three main OOXML file types. I want to have most of the same functionality but I wanted to add docx and pptx support. This is going to be a big project that will take a while but I want to work on it alot so if anyone else feels like contributing, the ECMA-376 document explaining the XML files is there and I will be adding to the github wiki to explain things in simpler and shorter terms.

**Comment by AshotN** [comment]
@romgrk Do you still plan on forking this?

**Comment by cherniavskii** [comment]
Hi @AshotN 
We haven't made the final decision yet.
But I want to clarify that our fork's purpose would be fixing the issues we're having at MUI X, not taking over `exceljs` maintenance.

**Comment by alexnguyen98** [comment]
There seems to be no commitments yet so @romgrk i think a fork should definitely exist

**Comment by par333k** [comment]
I've been working as a software engineer for seven years at a company that processes and handles Korean data. Because we've been using this library internally, I've personally forked and managed simple modifications. As someone who has always been a user of the open source ecosystem, I'm wondering if it's okay to move this project to another repository and continue working on resolving the pending issues. Since there's currently no communication, I'd like to find a solution.

**Comment by protobi-pieter** [comment]
@romgrk  We also  have a fork of this repo https://github.com/protobi/exceljs that we actively maintain and occasionally commission professional dev expert extensions (key issue is pivot table support).  We'd be happy to collaborate.

**Comment by protobi-pieter** [comment]
To all following this repo how many could contribute financially to its development?
Presumably Excel is for commercial use with funded budgets, not just kids subsisting on their parent's allowance.
I think one model is we can get a good efficient professional developer (we have one in mind) who is up to speed on this repo specifically and OpenXML generally and when we need a key feature implemented or pull request accepted we can commission that as a paid project with a feature spec, statement of work, test suite, and timeline.

**Comment by pcorpet** [comment]
There are now many forks that I know of:
- the one by the MUI team (@romgrk): https://www.npmjs.com/package/@mui/x-internal-exceljs-fork
- the one by protobi  (@protobi-pieter) : https://www.npmjs.com/package/@protobi/exceljs
- one by getnuvo (maybe @Elecweb): https://www.npmjs.com/package/nuvo-exceljs
- one by @rmartin93 https://www.npmjs.com/package/@rmartin93/exceljs-fork
- one by @alienkarma https://www.npmjs.com/package/@alienkarma/exceljs
- one by DevExtreme (@dmlvr) https://www.npmjs.com/package/devextreme-exceljs-fork

Thanks all for taking it up and start pushing new packages in npm. However having different forks make us include the original library several times in our app... Would one of you volunteer to take on the "main" fork, while others would be ok to stop their fork and fallback on the main one?

**Comment by romgrk** [comment]
We don't have the bandwidth to take over the ownership of the main fork, the fork we maintain is meant to fix the issues we had with it. I feel like the protobi fork might be a good candidate to continue the project, if the protobi team is willing to consider their fork as a permanent one (right now it's clearly stated to be a temporary fork) we'd definitely consider switching to that and adding our contributions there.

**Comment by xcybplx** [comment]
Another fork, which I know is not what this thread needs — so, briefly, what is different and what is not.

`exceljs-community` continues from 4.4.0 as a drop-in replacement: same API, migration is one line in `package.json`. 4.5.0 is published.

What it fixes so far is small and specific: #1328, where the streaming reader returned unresolved `{sharedString: N}` references instead of values when a worksheet is stored before `sharedStrings.xml`. The cause was in the stream iterator pausing around each `yield`; two other reader failures shared it.

What is different from the other forks is not the code — @romgrk, @protobi-pieter and others have fixed more than I have. It is that this one exists to be maintained rather than to solve one company's problem, and the promises are deliberately small enough to keep: a security release within 30 days, a response to issues within 14 days, no promises about features, and — the point of the exercise — if maintenance stops, it will be said out loud instead of leaving people to guess for two years.

No claim that this is the successor. If protobi or anyone else takes that role, better for everyone.

**Comment by gameroman** [comment]
I'd recommend trying out one of theseforks

https://github.com/excel-js/exceljs

https://github.com/cjnoname/excelts

https://github.com/protobi/exceljs

You can also see a [comparison on npmx](https://npmx.dev/compare?packages=exceljs,@cj-tech-master/excelts,@excel.js/exceljs,@protobi/exceljs)
